client.go

```diff
@@ -55,12 +52,28 @@ func NewClient(deployFile string, targetName TargetName, isLocalTest bool) (*Mul
 		return nil, err
 	}
 
+	appDir := path.Dir(deployFile)
+
 	// Create all SingleServerClients
 	clients := []Client{}
 	for _, target := range app.Targets(targetName) {
 		localPort := target.Base
+		var serverChannel ServerChannel
+
 		if !isLocalTest {
-			localPort = setupChannel(target.Base, target.SshPort, target.Ssh)
+			localPort = getFreeLocalPort()
+			serverChannel, err = NewSshChannel(
+				target.Base,
+				localPort,
+				target.SshPort,
+				target.Ssh,
+				newCommandRunner(appDir),
+			)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			serverChannel = NewLocalChannel(newCommandRunner(appDir))
```

Comment:
Sure, but still worried about against dev vs prod codepath divergence.  Can the tests have a flag so you can rerun them with sshd as a "final test"?

I like the sound of reusing ssh session, we should make that a thing.  Like, internally, you construct a client that keeps an ssh session and you then just send commands to it.  Which can then be easily wrapped in a REPL.  I reckin' we should do that.

Also, since for multiple servers -> multiple ssh connections... i think we need tests for these kinds of complexities.